src/PhotosphereViewer.tsx

```diff
@@ -141,13 +141,17 @@ export interface PhotosphereViewerProps {
   currentPS: string;
   onChangePS: (id: string) => void;
   onViewerClick?: (pitch: number, yaw: number) => void;
+  isEditorMode: boolean;
+  onRemoveHotspot: (hotspotToRemove: string) => void;
```

Comment:
The viewer mode won't have a delete button, so the `onRemoveHotspot` field should be optional. Then `isEditorMode` won't be necessary because you can just check that `onRemoveHotspot !== undefined`.